runtime/js/effect.js

```diff
@@ -142,12 +142,13 @@ function caml_perform_effect(eff, cont, last, k0) {
 }
 
 //Provides: caml_alloc_stack
-//Requires: caml_pop_fiber, caml_fiber_stack, caml_call_gen, caml_stack_check_depth, caml_trampoline_return
+//Requires: caml_pop_fiber, caml_call_gen, caml_stack_check_depth, caml_trampoline_return
 //If: effects
 //Version: >= 5.0
 function caml_alloc_stack(hv, hx, hf) {
+  var handlers = [0, hv, hx, hf];
   function call(i, x) {
-    var f = caml_fiber_stack.h[i];
+    var f = handlers[i];
```

Comment:
With the update to `caml_continuation_use_and_update_handler_noexc`, `caml_fiber_stack.h === handlers` seems to always hold.

Reply:
reverted and lifted outside